core/store/modules/cart/actions.ts

```diff
@@ -416,46 +416,50 @@ const actions: ActionTree<CartState, RootState> = {
     }
   },
   removeCoupon (context) {
-    if (rootStore.state.config.cart.synchronize_totals && (typeof navigator !== 'undefined' ? navigator.onLine : true)) {
-      context.dispatch('sync/execute', { url: rootStore.state.config.cart.deletecoupon_endpoint,
-        payload: {
-          method: 'POST',
-          headers: { 'Content-Type': 'application/json' },
-          mode: 'cors'
-        },
-        silent: true
-      }, { root: true }).then(task => {
-        if (task.result) {
-          context.dispatch('refreshTotals')
-        }
-      }).catch(e => {
-        console.error(e)
-      })
-    }
+    return new Promise((resolve, reject) => {
+      if (rootStore.state.config.cart.synchronize_totals && (typeof navigator !== 'undefined' ? navigator.onLine : true)) {
```

Comment:
I think we should ommit this online check. Then promis will just be rejected and we can handle error properly